### PR TITLE
Add icon to snackbar

### LINKED ISSUE #5436: Customer effort score: Show an icon in the snackbar
Part of #5259

There needs to be an icon in the snackbar - a pencil icon.

This needs to be added to the Snackbar component as an optional prop, and used by the CES snackbar prompt.

See https://woostartp2.wordpress.com/2020/10/16/customer-effort-score-design-i2/ for design discussion.

### PR DESCRIPTION
**Pull request by becdetat**
_Note that this is WIP, waiting on a final SVG for the pencil icon (@elizaan36)._

Fixes #5436 

This adds an optional icon to the snackbar, and uses the icon when the snackbar is displayed for CES.
image

*Testing instructions*

- `DELETE FROM wp_options WHERE option_name = 'woocommerce_ces_shown_for_actions'`
- Open a product for editing
- Edit the product and click Update
- Wait for a while and the snackbar should appear
- Make sure the snackbar has the icon and is aligned correctly

### CODE REVIEW

File: client/customer-effort-score-tracks/customer-effort-score-tracks.js [hunk 1]
```diff
@@ -83,6 +84,15 @@ function CustomerEffortScoreTracks( {
 			trackCallback={ trackCallback }
 			label={ label }
 			openedCallback={ openedCallback }
+			icon={
+				<span
+					style={ { height: 21, width: 21 } }
+					role="img"
+					aria-label={ __( 'Pencil icon', 'woocommerce-admin' ) }
+				>
+					✏️
```
Comment (becdetat) [inline client/customer-effort-score-tracks/customer-effort-score-tracks.js:93]: This is a pencil emoji, honest

File: packages/customer-effort-score/src/index.js [hunk 2]
```diff
@@ -6,27 +6,26 @@ import PropTypes from 'prop-types';
...
- * @param {Object}   props               Component props.
- * @param {Function} props.trackCallback Function to call when the results should be tracked.
- * @param {string}   props.label         The label displayed in the modal.
- * @param {Function} props.createNotice  Create a notice (snackbar).
+ * @param {Object}   props                Component props.
+ * @param {Function} props.trackCallback  Function to call when the results should be tracked.
+ * @param {string}   props.label          The label displayed in the modal.
+ * @param {Function} props.createNotice   Create a notice (snackbar).
  * @param {Function} props.openedCallback Function to call when the modal is opened.
+ * @param {string}   props.icon           Icon (React component) to be shown on the notice.
```
Comment (mattsherman) [inline packages/customer-effort-score/src/index.js:21]: Incorrect param type. Should be `Object`.

File: packages/customer-effort-score/src/index.js [hunk 3]
```diff
@@ -80,7 +83,15 @@ CustomerEffortScore.propTypes = {
...
-	createNotice: PropTypes.func,
+	createNotice: PropTypes.func.isRequired,
+	/**
+	 * Callback executed when the modal is opened.
+	 */
+	openedCallback: PropTypes.func,
+	/**
+	 * Icon (React component) to be displayed.
+	 */
+	icon: PropTypes.object,
```
Comment (mattsherman) [inline packages/customer-effort-score/src/index.js:94]: I think this should be `PropTypes.element`.

**Review by mattsherman — APPROVED** [review]
Tested and works as expected.

Just some minor doc updates requested. I would be totally cool with you merging this after making those updates, without another round of review.

**Comment by becdetat** [comment]
Thanks @mattsherman :-) I've made the suggested changes (good catch) and I'll merge this now.

**Pull request merged**